app/src/main/java/com/zulip/android/activities/ZulipActivity.java

```diff
@@ -237,8 +239,7 @@ protected void onCreate(Bundle savedInstanceState) {
             return;
         }
         this.logged_in = true;
-        notifications = new Notifications(this);
-        notifications.register();
+        registerGCM();
```

Comment:
Needs to 'unregister' GCM after logout, including delete the saved google_client_id preference .